pkg/controller/workload/job/job_controller.go

```diff
@@ -370,8 +371,24 @@ func (r *JobReconciler) ensureAtMostOneWorkload(ctx context.Context, job *batchv
 	return match, nil
 }
 
-func ConstructWorkloadFor(job *batchv1.Job, scheme *runtime.Scheme) (*kueue.QueuedWorkload, error) {
-	w := &kueue.QueuedWorkload{
+func ConstructWorkloadFor(ctx context.Context, client client.Client,
+	job *batchv1.Job, scheme *runtime.Scheme) (w *kueue.QueuedWorkload, err error) {
+	var p int32
+	pcName := job.Spec.Template.Spec.PriorityClassName
```

Comment:
What will be do for workloads that have multiple pod specs?

Should we do this in the queued_workload controller instead?

Reply:
I rarely see people setting different priorities in a single workload. But in this case, we need a fixed treatment and clear understanding for the user: like choosing the highest one as the priority of the workload?

Reply:
> What will be do for workloads that have multiple pod specs?

It is up to the custom workloads to decide from where to set it (e.g., based on the driver).

> Should we do this in the queued_workload controller instead?

how? queuedworkload_controller is not aware of the custom workload CRD.


That makes me think, since order is done at the ClusterQueue level and Queue is simply a pointer to it; what if we have the priority on the Queue, and workloads simply inherit it from the Queue they got submitted to, and so users don't need to set it on the custom workload directly and we avoid those issues altogether.

Reply:
>> Should we do this in the queued_workload controller instead?
>
> how? queuedworkload_controller is not aware of the custom workload CRD.

I'm thinking that workloads should have a priority independent of the pod priority, although it might be confusing. Also that would mean that users have to set an annotation for it in the Job or something like that. Having it in the Queue is certainly a cleaner option.

Reply:
I see. Do you know what they do with it? Perhaps insert it in the pod specs?